### PR TITLE
layers: Remove unused ValidateViConsistency

### PR DESCRIPTION
**Pull request by spencer-lunarg**
I realized that `ValidateViConsistency` is not being checked because there is already 

> VUID-VkPipelineVertexInputStateCreateInfo-pVertexBindingDescriptions-00616
>
> All elements of pVertexBindingDescriptions must describe distinct binding numbers

That is validating this

Also while at it, took the `vkCmdSetVertexInputEXT` logic and made it consistent and improved the error message

### CODE REVIEW

File: layers/parameter_validation_utils.cpp [hunk 1]
```diff
@@ -8354,31 +8354,34 @@ bool StatelessValidation::manual_PreCallValidateCmdSetVertexInputEXT(
...
-                    skip |= LogError(device, "VUID-vkCmdSetVertexInputEXT-pVertexBindingDescriptions-04794",
-                                     "vkCmdSetVertexInputEXT(): binding description for binding %" PRIu32 " already specified",
-                                     binding_value);
-                }
+    // check for distinct values
+    {
+        layer_data::unordered_set<uint32_t> vertex_bindings(vertexBindingDescriptionCount);
+        for (uint32_t i = 0; i < vertexBindingDescriptionCount; ++i) {
+            const uint32_t binding = pVertexBindingDescriptions[i].binding;
+            auto const &binding_it = vertex_bindings.find(binding);
```
Comment (juan-lunarg) [inline layers/parameter_validation_utils.cpp:8362]: `find` doesn't return a const reference

File: layers/parameter_validation_utils.cpp [hunk 2]
```diff
@@ -8354,31 +8354,34 @@ bool StatelessValidation::manual_PreCallValidateCmdSetVertexInputEXT(
...
-            for (uint32_t next_attribute = attribute + 1; next_attribute < vertexAttributeDescriptionCount; ++next_attribute) {
-                if (location == pVertexAttributeDescriptions[next_attribute].location) {
-                    skip |= LogError(device, "VUID-vkCmdSetVertexInputEXT-pVertexAttributeDescriptions-04795",
-                                     "vkCmdSetVertexInputEXT(): attribute description for location %" PRIu32 " already specified",
-                                     location);
-                }
+        layer_data::unordered_set<uint32_t> vertex_locations(vertexBindingDescriptionCount);
+        for (uint32_t i = 0; i < vertexBindingDescriptionCount; ++i) {
+            const uint32_t location = pVertexAttributeDescriptions[i].location;
+            auto const &location_it = vertex_locations.find(location);
```
Comment (juan-lunarg) [inline layers/parameter_validation_utils.cpp:8376]: `find` doesn't return a const reference

**Review by juan-lunarg — DISMISSED** [review]

File: layers/parameter_validation_utils.cpp (hunk 1 above)
Comment (ncesario-lunarg) [reply layers/parameter_validation_utils.cpp:8362]: I believe it is legal to assign an rvalue to a const ref and this _can_ (??) avoid copy construction, so I think this is fine, but it's quite possible I'm not remembering this detail correctly.

**Review by ncesario-lunarg — COMMENTED** [review]
LGTM once Juan's comments are resolved.

File: layers/parameter_validation_utils.cpp (hunk 1 above)
Comment (juan-lunarg) [reply layers/parameter_validation_utils.cpp:8362]: @ncesario-lunarg is correct. I forgot the nuance for `const` references. [Source FWIW](https://stackoverflow.com/questions/22845167/const-reference-and-lvalue)

**Review by juan-lunarg — DISMISSED** [review]

**Review by juan-lunarg — CHANGES_REQUESTED** [review]
```
[ctest] The following tests FAILED:
[ctest] 	594 - VkLayerTest.ValidateVertexInputDynamicStateEnabled (SEGFAULT)
[ctest] 	595 - VkLayerTest.ValidateVertexInputDynamicStateDivisor (SEGFAULT)
```

File: layers/parameter_validation_utils.cpp [hunk 3]
```diff
@@ -8354,31 +8354,34 @@ bool StatelessValidation::manual_PreCallValidateCmdSetVertexInputEXT(
...
-        for (uint32_t attribute = 0; attribute < vertexAttributeDescriptionCount - 1; ++attribute) {
-            uint32_t location = pVertexAttributeDescriptions[attribute].location;
-            for (uint32_t next_attribute = attribute + 1; next_attribute < vertexAttributeDescriptionCount; ++next_attribute) {
-                if (location == pVertexAttributeDescriptions[next_attribute].location) {
-                    skip |= LogError(device, "VUID-vkCmdSetVertexInputEXT-pVertexAttributeDescriptions-04795",
-                                     "vkCmdSetVertexInputEXT(): attribute description for location %" PRIu32 " already specified",
-                                     location);
-                }
+        layer_data::unordered_set<uint32_t> vertex_locations(vertexBindingDescriptionCount);
+        for (uint32_t i = 0; i < vertexBindingDescriptionCount; ++i) {
```
Comment (juan-lunarg) [inline layers/parameter_validation_utils.cpp:8374]:
```suggestion
        layer_data::unordered_set<uint32_t> vertex_locations(vertexAttributeDescriptionCount);
        for (uint32_t i = 0; i < vertexAttributeDescriptionCount; ++i) {
```

**Review by juan-lunarg — CHANGES_REQUESTED** [review]

**Review by juan-lunarg — APPROVED** [review]
LGTM

**Pull request merged**